AdyenComponents/Affirm/AffirmComponent.swift

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import UIKit
+
+/// A component that provides a form for Affirm payment.
+public final class AffirmComponent: AbstractPersonalInformationComponent, Observer {
+    
+    // MARK: - Items
+    
+    /// :nodoc:
+    private let personalDetailsHeaderItem: FormLabelItem
+    
+    /// :nodoc:
+    public let deliveryAddressToggleItem: FormToggleItem
+    
+    /// :nodoc:
+    public let deliveryAddressItem: FormAddressItem
+    
+    // MARK: - Properties
+    
+    private var isDeliveryAddressEnabled: Bool {
+        deliveryAddressToggleItem.value
+    }
+    
+    // MARK: - Initializers
+    
+    /// Initializes the Affirm component.
+    /// - Parameters:
+    ///   - paymentMethod: The Affirm payment method.
+    ///   - apiContext: The component's API context.
+    ///   - style: The component's style.
+    public init(paymentMethod: PaymentMethod,
+                apiContext: APIContext,
+                style: FormComponentStyle) {
+        personalDetailsHeaderItem = FormLabelItem(text: "", style: style.sectionHeader)
+        deliveryAddressToggleItem = FormToggleItem(style: style.toggle)
+        deliveryAddressItem = FormAddressItem(initialCountry: Locale.current.regionCode ?? "US", style: style.addressStyle)
+        
+        let fields: [PersonalInformation] = [
+            .custom(CustomFormItemInjector(item: personalDetailsHeaderItem.addingDefaultMargins())),
+            .firstName,
+            .lastName,
+            .email,
+            .phone,
+            .address,
+            .custom(CustomFormItemInjector(item: FormSpacerItem(numberOfSpaces: 16))),
+            .custom(CustomFormItemInjector(item: deliveryAddressToggleItem)),
+            .custom(CustomFormItemInjector(item: deliveryAddressItem)),
+            .custom(CustomFormItemInjector(item: FormSpacerItem(numberOfSpaces: 24)))
+        ]
```

Comment:
Lines 40-55 could be a private static function?

Reply:
I thought about it. You mean something like this right?

```
private static func getFields() -> [PersonalInformation] {
   ...
}
```

Reply:
yes, why not :)

Reply:
I thought about it but I'm afraid it's not possible since I need to have access to the first name, last name, email and phone fields from the parent class (`AbstractPersonalInformationComponent`). Cannot access them from a static method.